httpcore5/src/main/java/org/apache/hc/core5/http/io/entity/InputStreamEntity.java

```diff
@@ -31,87 +31,52 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 
+import org.apache.hc.core5.annotation.Contract;
+import org.apache.hc.core5.annotation.ThreadingBehavior;
 import org.apache.hc.core5.http.ContentType;
 import org.apache.hc.core5.util.Args;
 
 /**
- * A streamed, non-repeatable entity that obtains its content from
- * an {@link InputStream}.
+ * A streamed, non-repeatable entity that obtains its content from an {@link InputStream}.
  *
  * @since 4.0
  */
+@Contract(threading = ThreadingBehavior.IMMUTABLE_CONDITIONAL)
```

Comment:
I don't really see how an `InputStream` wrapper could be regarded as immutable, unless `IMMUTABLE_CONDITIONAL` specifically means "immutable" in the Java Memory Model sense (all final fields, not made visible to other threads during construction), which has significantly weaker guarantees than true immutability.

Reply:
@rschmitt This is why there is `IMMUTABLE` and `IMMUTABLE_CONDITIONAL` with much weaker guarantees.